doc/table.dox

```diff
@@ -158,17 +158,58 @@ namespace lsst { namespace afw { namespace table {
  *  discussed below).
  *
  *  @section afwTableFieldNames Field Names
- *  By convention, field names are all lowercase and have '.'-separated elements.  Only letters, numbers
- *  and periods should be used.  These rules are not enforced, but names that do not meet these requirements
- *  may not round-trip correctly in FITS (periods are converted to underscores in the FITS persistence
- *  layer, so we cannot distinguish between the two when we read tables from FITS).
- *  Schema provides extra functionality for names with period-separated elements; these elements can
- *  be accessed separately individually with the bracket operators.  More information on schema namespaces
- *  can be found in the Schema and SubSchema class documentation, and the testSchema.py unit test may
- *  also be a useful example.
- *
- *  Other field strings (documentation and units) are essentially arbitrary, but should not contain
- *  single quotes, as these may also confuse FITS parsers (even when escaped).
+ *
+ *  By convention, field names are CamelCase and have '_'-separated elements.
```

Comment:
Would it be more accurate to say (c)CamelCase (or some other way of emphasizing that the initial letter need not always be capitalized)?

Might it help to provide a list of examples that fall into the category of "element" in this case (no worries if you think the meaning of the word is clear/unambiguous in the context of code)?

Reply:
I'll just write "camel case" here and not imply anything with how its capitalized.  I think later uses are okay because they're right above the section saying that starting with lowercase is okay.

I think I prefer having the list of element examples in the next paragraph, just to keep this paragraph simple.